internal/plugin/github.go

```diff
@@ -5,22 +5,45 @@ import (
 	"io"
 	"net/http"
 	"net/url"
+	"regexp"
+	"strings"
 
+	"github.com/samber/lo"
 	"go.jetpack.io/devbox/internal/boxcli/usererr"
 	"go.jetpack.io/devbox/internal/cachehash"
 	"go.jetpack.io/devbox/nix/flake"
 )
 
 type githubPlugin struct {
-	ref flake.Ref
+	ref  flake.Ref
+	name string
+}
+
+// Github only allows alphanumeric, hyphen, underscore, and period in repo names.
+// but we clean up just in case.
+var githubNameRegexp = regexp.MustCompile("[^a-zA-Z0-9-_.]+")
+
+func newGithubPlugin(ref flake.Ref) *githubPlugin {
+	plugin := &githubPlugin{ref: ref}
+	// For backward compatibility, we don't strictly require name to be present
+	// in github plugins. If it's missing, we just use the directory as the name.
+	name, _ := getPluginNameFromContent(plugin)
```

Comment:
lets do a followup that checks the error to be `ErrNameOmitted` since right now it conflates malformed name with missing name.

Not blocking because I think the change could be a bit involved since you'd need to propagate the error from this function...

Reply:
actually, is distinguishing this scenario important for this code flow? I'm now having doubts....